gpii/node_modules/dbOperation/src/DbDataStoreUtils.js

```diff
@@ -0,0 +1,560 @@
+ * GPII key record, or an error if `gpiiKey` or `gpiiKeyData` is not provided or `gpiiKeyData.type` is not
+ * the doc type for gpiiKey.
+ */
+gpii.dbOperation.dbDataStore.updateGpiiKey = function (saveDataSource, gpiiKey, gpiiKeyData) {
+    var promiseTogo = fluid.promise();
+    var data, error;
+
+    if (!gpiiKey) {
+        error = gpii.dbOperation.composeError(gpii.dbOperation.errors.missingInput, {fieldName: "gpiiKey"});
+        promiseTogo.reject(error);
+    } else if (!gpiiKeyData) {
+        error = gpii.dbOperation.composeError(gpii.dbOperation.errors.missingDoc, {docName: gpii.dbOperation.docTypes.gpiiKey});
+        promiseTogo.reject(error);
+    } else {
+        data = {
+            type: gpiiKeyData.type,
+            schemaVersion: gpiiKeyData.schemaVersion,
+            prefsSafeId: gpiiKeyData.prefsSafeId,
+            prefsSetId: gpiiKeyData.prefsSetId,
+            revoked: gpiiKeyData.revoked,
+            revokedReason: gpiiKeyData.revokedReason,
+            timestampCreated: gpiiKeyData.timestampCreated,
+            timestampUpdated: gpii.dbOperation.getCurrentTimestamp(),
+            timestampRevoked: gpiiKeyData.timestampRevoked
+        };
+
+        promiseTogo = gpii.dbOperation.dbDataStore.updateRecord(saveDataSource, gpii.dbOperation.docTypes.gpiiKey, gpiiKey, data);
+    }
+    return promiseTogo;
+};
+
+// General Preferences Safes Functions
+// -----------------------------------
+
+/**
+ * Transform the data in CouchDB form to a more understandable structure
+ * @param data {Object} Contains GPII key and preferences safe information associated with a GPII key
+ * An input example:
+ * {
+ *     key: {String},   // GPII key
+ *     id: {String},    // GPII key
+ *     value: {
+ *         _id: {String},      // prefs Safe id
+ *         gpiiKey: {
+ *             type: {String},
+ *             schemaVersion: {String},
+ *             prefsSafeId: {String},
+ *             prefsSetId: {String},
+ *             revoked: {Boolean},
+ *             revokedReason: {String},
+ *             timestampCreated: {Date},
+ *             timestampUpdated: {Date},
+ *             _id: {String},
+ *             _rev: {String}
+ *         }
+ *     },
+ *     doc: {
+ *         type: {String},
+ *         schemaVersion: {String},
+ *         name: {String},
+ *         password: {String},
+ *         email: {String},
+ *         preferences: {Object},
+ *         timestampCreated: {Date},
+ *         timestampUpdated: {Date},
+ *         _id: {String},
+ *         _rev: {String}
+ *     }
+ * }
+ * @return {Object} An object in the structure:
+ * {
+ *     gpiiKey: {String},
+ *     gpiiKeyDetails: {Object},
+ *     prefsSafe: {Object}
+ * }
+ */
+gpii.dbOperation.dbDataStore.findPrefsSafeByGpiiKeyPostProcess = function (data) {
+    var result;
+
+    if (data && data.doc && data.value) {
+        result = {};
+        fluid.set(result, "gpiiKey", data.key);
```

Comment:
I don't think that fluid.set is necessary here since everything is a direct property.
Why not just 
result = {
    "gpiiKey": data.key
...
}
etc.